### PR TITLE
Support Intel MKL CBLAS backend.

### PR DESCRIPTION
**Pull request by evilmucedin**
Intel MKL library is known as the best CBLAS implementation for Intel platforms. This changelist adds support of Intel MKL backend for fully-connected layer. An existing test case for NNPACK was extended for the new backend type.

### CODE REVIEW

**Comment by bhack** [comment]
Thank you for the contrib. What we are really missing here other then high performance backend it is a graph optimization pass. We have evaluated different solutions one year ago like [TVM/NNVM](https://docs.tvm.ai/dev/index.html)

**Comment by bhack** [comment]
I we know its good and we appreciate the PR. Was just a meta reflection.

**Comment by beru** [comment]
Please rebase to the master branch.

**Comment by bhack** [comment]
What about mkl vs mkl-dnn: https://github.com/intel/mkl-dnn?

>DNN functionality optimized for Intel architecture is also included in Intel(R) Math Kernel Library (Intel(R) MKL). API in this implementation is not compatible with Intel MKL-DNN and does not include certain new and experimental features.

File: tiny_dnn/core/kernels/fully_connected_op_intel_mkl.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      memset(output, 0, sizeof(float_t) * out_size);
+    else
+      memcpy(output, bias.data(), sizeof(float_t) * out_size);
+#ifdef CNN_USE_DOUBLE
+    cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+                alpha, input, in_size, weight, out_size, beta, output, out_size);
+#else
+    cblas_sgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+                alpha, input, in_size, weight, out_size, beta, output, out_size);
+  }
```
Comment (beru) [inline tiny_dnn/core/kernels/fully_connected_op_intel_mkl.h]: Using Batch GEMM might give more speed benefits.
https://software.intel.com/en-us/articles/introducing-batch-gemm-operations

**Pull request merged**